Code/ETCDLibrary/src/main/java/ie/tcd/wayfinder/etcd/ETCDLibrary.java

```diff
@@ -0,0 +1,91 @@
+package ie.tcd.wayfinder.etcd;
+
+import ie.tcd.wayfinders.restLibrary.Library;
+
+import java.io.IOException;
+import java.util.Optional;
+
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.util.EntityUtils;
+
+public class ETCDLibrary {
+
+	private static String baseUrl = "http://35.246.14.12:2379/v2/keys/";
+		
+	public static String ReadDirectory(String directoryName) throws IOException {
+		
+		String url = (baseUrl + directoryName).trim();
+		HttpResponse response = Library.GET(url, Optional.empty());
+		
+		 HttpEntity responseEntity = response.getEntity();
+	     String responseXml = EntityUtils.toString(responseEntity);
```

Comment:
Is it `responseXml`. AFAIK, we get a JSON. And why should we parse it as a `String`?